### PR TITLE
Adjust distributuon policy.

### PR DESCRIPTION
**Pull request by ipanova**
[noisuse]

### CODE REVIEW

File: pulp_container/app/viewsets.py [hunk 1]
```diff
@@ -995,7 +995,19 @@ class ContainerDistributionViewSet(DistributionViewSet):
...
+                "condition": "has_namespace_model_perms",
+            },
+            {
+                "action": ["create"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_namespace_perms:container.add_containerdistribution",
+            },
+            {
+                "action": ["create"],
```
Comment (ipanova) [inline pulp_container/app/viewsets.py:1007]: allows to create distribution and namespace whenever username matches

File: pulp_container/app/viewsets.py [hunk 2]
```diff
@@ -995,7 +995,19 @@ class ContainerDistributionViewSet(DistributionViewSet):
                 "action": ["create"],
                 "principal": "authenticated",
                 "effect": "allow",
-                "condition": "has_model_perms:container.add_containerdistribution",
+                "condition": "has_namespace_model_perms",
+            },
+            {
+                "action": ["create"],
```
Comment (ipanova) [inline pulp_container/app/viewsets.py:1001]: condition to match when namespace exists

File: pulp_container/app/viewsets.py [hunk 3]
```diff
@@ -995,7 +995,19 @@ class ContainerDistributionViewSet(DistributionViewSet):
                 "action": ["create"],
                 "principal": "authenticated",
                 "effect": "allow",
-                "condition": "has_model_perms:container.add_containerdistribution",
+                "condition": "has_namespace_model_perms",
```
Comment (ipanova) [inline pulp_container/app/viewsets.py:998]: condition to match when namespace does not exist

**Review by mdellweg — APPROVED** [review]
LGTM.

**Comment by ipanova** [comment]
I don't think these changes require a migration, so marking as ready for re-review.

File: pulp_container/app/access_policy.py [hunk 4]
```diff
@@ -54,6 +68,22 @@ def is_private(self, request, view, action):
...
+        """
+        Check if the namespace in the request matches the username.
+        """
+        base_path = request.data.get("base_path")
+        namespace = base_path.split("/")[0]
+        return namespace == request.user.username
+
+    def has_namespace_model_perms(self, request, view, action):
+        """
+        Check ifthe user can create namespaces.
```
Comment (mdellweg) [inline pulp_container/app/access_policy.py:81]:
```suggestion
        Check if the user can create namespaces.
```

**Review by mdellweg — APPROVED** [review]

**Review by goosemania — APPROVED** [review]

**Pull request merged**